core/src/include/Xios.hpp

```diff
@@ -30,12 +30,37 @@ namespace Nextsim {
 void enableXios();
 
 class Xios : public Configured<Xios> {
-public:
+private:
+    static Xios* instancePtr;
```

Comment:
Avoid static members in classes. Wrap this in a function to allow it to at least be lazily constructed. See for example [Configurator.hpp:185](https://github.com/nextsimhub/nextsimdg/blob/100e2b742713ce6a809bc2d7fa9386864a1f1852/core/src/include/Configurator.hpp#L185)

Making the pointer a static variable in `getInstance()` would achieve this.

Reply:
Thanks for this suggestion and for pointing to an example @timspainNERSC. This is a much neater implementation. Done in 10d6207d42813c5009e7a69bba47c07bfd05f110.